File: src/restricted-extras-ktextaddons/libimportabstractautocorrection.cpp
#include <QHash>
#include <QSet>
#include <QString>
#define WORKAROUND_INNER_CLASS_DEFINITION_TextAutoCorrectionCore__AutoCorrectionUtils__TypographicQuotes
#define WORKAROUND_INNER_CLASS_DEFINITION_TextAutoCorrectionCore__ImportAbstractAutocorrection
#include <importabstractautocorrection.h>
#include "libimportabstractautocorrection.h"
#include "libimportabstractautocorrection.hxx"

TextAutoCorrectionCore__ImportAbstractAutocorrection* TextAutoCorrectionCore__ImportAbstractAutocorrection_new() {
    return new VirtualTextAutoCorrectionCoreImportAbstractAutocorrection();
}

TextAutoCorrectionCore__ImportAbstractAutocorrection* TextAutoCorrectionCore__ImportAbstractAutocorrection_new2(const TextAutoCorrectionCore__ImportAbstractAutocorrection* param1) {
    return new VirtualTextAutoCorrectionCoreImportAbstractAutocorrection(*param1);
}

bool TextAutoCorrectionCore__ImportAbstractAutocorrection_Import(TextAutoCorrectionCore__ImportAbstractAutocorrection* self, const libqt_string fileName, libqt_string errorMessage, int loadAttribute) {
    QString fileName_QString = QString::fromUtf8(fileName.data, fileName.len);
    QString errorMessage_QString = QString::fromUtf8(errorMessage.data, errorMessage.len);
    return self->import(fileName_QString, errorMessage_QString, static_cast<TextAutoCorrectionCore::ImportAbstractAutocorrection::LoadAttribute>(loadAttribute));
}

libqt_list /* set of libqt_string */ TextAutoCorrectionCore__ImportAbstractAutocorrection_UpperCaseExceptions(const TextAutoCorrectionCore__ImportAbstractAutocorrection* self) {
    QSet<QString> _ret = self->upperCaseExceptions();
    // Convert QSet<> from C++ memory to manually-managed C memory
    libqt_string* _arr = static_cast<libqt_string*>(malloc(sizeof(libqt_string) * _ret.size()));
    int _ctr = 0;
    QSetIterator<QString> _itr(_ret);
    while (_itr.hasNext()) {
        auto _sv_ret = _itr.next();
        // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
        QByteArray _sv_b = _sv_ret.toUtf8();
        libqt_string _sv_str;
        _sv_str.len = _sv_b.length();
        _sv_str.data = static_cast<const char*>(malloc(_sv_str.len + 1));
        memcpy((void*)_sv_str.data, _sv_b.data(), _sv_str.len);
        ((char*)_sv_str.data)[_sv_str.len] = '\0';
        _arr[_ctr++] = _sv_str;
    }
    libqt_list _out;
    _out.len = _ret.size();
    _out.data = static_cast<void*>(_arr);
    return _out;
}

libqt_list /* set of libqt_string */ TextAutoCorrectionCore__ImportAbstractAutocorrection_TwoUpperLetterExceptions(const TextAutoCorrectionCore__ImportAbstractAutocorrection* self) {
    QSet<QString> _ret = self->twoUpperLetterExceptions();
    // Convert QSet<> from C++ memory to manually-managed C memory
    libqt_string* _arr = static_cast<libqt_string*>(malloc(sizeof(libqt_string) * _ret.size()));
    int _ctr = 0;
    QSetIterator<QString> _itr(_ret);
    while (_itr.hasNext()) {
        auto _sv_ret = _itr.next();
        // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
        QByteArray _sv_b = _sv_ret.toUtf8();
        libqt_string _sv_str;
        _sv_str.len = _sv_b.length();
        _sv_str.data = static_cast<const char*>(malloc(_sv_str.len + 1));
        memcpy((void*)_sv_str.data, _sv_b.data(), _sv_str.len);
        ((char*)_sv_str.data)[_sv_str.len] = '\0';
        _arr[_ctr++] = _sv_str;
    }
    libqt_list _out;
    _out.len = _ret.size();
    _out.data = static_cast<void*>(_arr);
    return _out;
}

libqt_map /* of libqt_string to libqt_string */ TextAutoCorrectionCore__ImportAbstractAutocorrection_AutocorrectEntries(const TextAutoCorrectionCore__ImportAbstractAutocorrection* self) {
    QHash<QString, QString> _ret = self->autocorrectEntries();
    // Convert QHash<> from C++ memory to manually-managed C memory
    libqt_string* _karr = static_cast<libqt_string*>(malloc(sizeof(libqt_string) * _ret.size()));
    libqt_string* _varr = static_cast<libqt_string*>(malloc(sizeof(libqt_string) * _ret.size()));
    int _ctr = 0;
    for (auto _itr = _ret.keyValueBegin(); _itr != _ret.keyValueEnd(); ++_itr) {
        auto _hashkey_ret = _itr->first;
        // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
        QByteArray _hashkey_b = _hashkey_ret.toUtf8();
        libqt_string _hashkey_str;
        _hashkey_str.len = _hashkey_b.length();
        _hashkey_str.data = static_cast<const char*>(malloc(_hashkey_str.len + 1));
        memcpy((void*)_hashkey_str.data, _hashkey_b.data(), _hashkey_str.len);
        ((char*)_hashkey_str.data)[_hashkey_str.len] = '\0';
        _karr[_ctr] = _hashkey_str;
        auto _hashval_ret = _itr->second;
        // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
        QByteArray _hashval_b = _hashval_ret.toUtf8();
        libqt_string _hashval_str;
        _hashval_str.len = _hashval_b.length();
        _hashval_str.data = static_cast<const char*>(malloc(_hashval_str.len + 1));
        memcpy((void*)_hashval_str.data, _hashval_b.data(), _hashval_str.len);
        ((char*)_hashval_str.data)[_hashval_str.len] = '\0';
        _varr[_ctr] = _hashval_str;
        _ctr++;
    }
    libqt_map _out;
    _out.len = _ret.size();
    _out.keys = static_cast<void*>(_karr);
    _out.values = static_cast<void*>(_varr);
    return _out;
}

libqt_map /* of libqt_string to libqt_string */ TextAutoCorrectionCore__ImportAbstractAutocorrection_SuperScriptEntries(const TextAutoCorrectionCore__ImportAbstractAutocorrection* self) {
    QHash<QString, QString> _ret = self->superScriptEntries();
    // Convert QHash<> from C++ memory to manually-managed C memory
    libqt_string* _karr = static_cast<libqt_string*>(malloc(sizeof(libqt_string) * _ret.size()));
    libqt_string* _varr = static_cast<libqt_string*>(malloc(sizeof(libqt_string) * _ret.size()));
    int _ctr = 0;
    for (auto _itr = _ret.keyValueBegin(); _itr != _ret.keyValueEnd(); ++_itr) {
        auto _hashkey_ret = _itr->first;
        // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
        QByteArray _hashkey_b = _hashkey_ret.toUtf8();
        libqt_string _hashkey_str;
        _hashkey_str.len = _hashkey_b.length();
        _hashkey_str.data = static_cast<const char*>(malloc(_hashkey_str.len + 1));
        memcpy((void*)_hashkey_str.data, _hashkey_b.data(), _hashkey_str.len);
        ((char*)_hashkey_str.data)[_hashkey_str.len] = '\0';
        _karr[_ctr] = _hashkey_str;
        auto _hashval_ret = _itr->second;
        // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
        QByteArray _hashval_b = _hashval_ret.toUtf8();
        libqt_string _hashval_str;
        _hashval_str.len = _hashval_b.length();
        _hashval_str.data = static_cast<const char*>(malloc(_hashval_str.len + 1));
        memcpy((void*)_hashval_str.data, _hashval_b.data(), _hashval_str.len);
        ((char*)_hashval_str.data)[_hashval_str.len] = '\0';
        _varr[_ctr] = _hashval_str;
        _ctr++;
    }
    libqt_map _out;
    _out.len = _ret.size();
    _out.keys = static_cast<void*>(_karr);
    _out.values = static_cast<void*>(_varr);
    return _out;
}

TextAutoCorrectionCore__AutoCorrectionUtils__TypographicQuotes* TextAutoCorrectionCore__ImportAbstractAutocorrection_TypographicSingleQuotes(const TextAutoCorrectionCore__ImportAbstractAutocorrection* self) {
    return new TextAutoCorrectionCore::AutoCorrectionUtils::TypographicQuotes(self->typographicSingleQuotes());
}

TextAutoCorrectionCore__AutoCorrectionUtils__TypographicQuotes* TextAutoCorrectionCore__ImportAbstractAutocorrection_TypographicDoubleQuotes(const TextAutoCorrectionCore__ImportAbstractAutocorrection* self) {
    return new TextAutoCorrectionCore::AutoCorrectionUtils::TypographicQuotes(self->typographicDoubleQuotes());
}

int TextAutoCorrectionCore__ImportAbstractAutocorrection_MaxFindStringLenght(const TextAutoCorrectionCore__ImportAbstractAutocorrection* self) {
    return self->maxFindStringLenght();
}

int TextAutoCorrectionCore__ImportAbstractAutocorrection_MinFindStringLenght(const TextAutoCorrectionCore__ImportAbstractAutocorrection* self) {
    return self->minFindStringLenght();
}

// Base class handler implementation
bool TextAutoCorrectionCore__ImportAbstractAutocorrection_SuperImport(TextAutoCorrectionCore__ImportAbstractAutocorrection* self, const libqt_string fileName, libqt_string errorMessage, int loadAttribute) {
    auto* vtextautocorrectioncoreimportabstractautocorrection = dynamic_cast<VirtualTextAutoCorrectionCoreImportAbstractAutocorrection*>(self);
    QString fileName_QString = QString::fromUtf8(fileName.data, fileName.len);
    QString errorMessage_QString = QString::fromUtf8(errorMessage.data, errorMessage.len);
    if (vtextautocorrectioncoreimportabstractautocorrection && vtextautocorrectioncoreimportabstractautocorrection->isVirtualTextAutoCorrectionCoreImportAbstractAutocorrection) {
        vtextautocorrectioncoreimportabstractautocorrection->setTextAutoCorrectionCore__ImportAbstractAutocorrection_Import_IsBase(true);
        return vtextautocorrectioncoreimportabstractautocorrection->import(fileName_QString, errorMessage_QString, static_cast<TextAutoCorrectionCore::ImportAbstractAutocorrection::LoadAttribute>(loadAttribute));
    } else {
        return ((VirtualTextAutoCorrectionCoreImportAbstractAutocorrection*)self)->import(fileName_QString, errorMessage_QString, static_cast<TextAutoCorrectionCore::ImportAbstractAutocorrection::LoadAttribute>(loadAttribute));
    }
}

// Auxiliary method to allow providing re-implementation
void TextAutoCorrectionCore__ImportAbstractAutocorrection_OnImport(TextAutoCorrectionCore__ImportAbstractAutocorrection* self, intptr_t slot) {
    auto* vtextautocorrectioncoreimportabstractautocorrection = dynamic_cast<VirtualTextAutoCorrectionCoreImportAbstractAutocorrection*>(self);
    if (vtextautocorrectioncoreimportabstractautocorrection && vtextautocorrectioncoreimportabstractautocorrection->isVirtualTextAutoCorrectionCoreImportAbstractAutocorrection) {
        vtextautocorrectioncoreimportabstractautocorrection->setTextAutoCorrectionCore__ImportAbstractAutocorrection_Import_Callback(reinterpret_cast<VirtualTextAutoCorrectionCoreImportAbstractAutocorrection::TextAutoCorrectionCore__ImportAbstractAutocorrection_Import_Callback>(slot));
    }
}

// Derived class handler implementation
void TextAutoCorrectionCore__ImportAbstractAutocorrection_OperatorAssign(TextAutoCorrectionCore__ImportAbstractAutocorrection* self, const TextAutoCorrectionCore__ImportAbstractAutocorrection* param1) {
    auto* vtextautocorrectioncoreimportabstractautocorrection = dynamic_cast<VirtualTextAutoCorrectionCoreImportAbstractAutocorrection*>(self);
    if (vtextautocorrectioncoreimportabstractautocorrection && vtextautocorrectioncoreimportabstractautocorrection->isVirtualTextAutoCorrectionCoreImportAbstractAutocorrection) {
        vtextautocorrectioncoreimportabstractautocorrection->operator=(*param1);
    } else {
        ((VirtualTextAutoCorrectionCoreImportAbstractAutocorrection*)self)->operator=(*param1);
    }
}

// Base class handler implementation
void TextAutoCorrectionCore__ImportAbstractAutocorrection_SuperOperatorAssign(TextAutoCorrectionCore__ImportAbstractAutocorrection* self, const TextAutoCorrectionCore__ImportAbstractAutocorrection* param1) {
    auto* vtextautocorrectioncoreimportabstractautocorrection = dynamic_cast<VirtualTextAutoCorrectionCoreImportAbstractAutocorrection*>(self);
    if (vtextautocorrectioncoreimportabstractautocorrection && vtextautocorrectioncoreimportabstractautocorrection->isVirtualTextAutoCorrectionCoreImportAbstractAutocorrection) {
        vtextautocorrectioncoreimportabstractautocorrection->setTextAutoCorrectionCore__ImportAbstractAutocorrection_OperatorAssign_IsBase(true);
        vtextautocorrectioncoreimportabstractautocorrection->operator=(*param1);
    } else {
        ((VirtualTextAutoCorrectionCoreImportAbstractAutocorrection*)self)->operator=(*param1);
    }
}

// Auxiliary method to allow providing re-implementation
void TextAutoCorrectionCore__ImportAbstractAutocorrection_OnOperatorAssign(TextAutoCorrectionCore__ImportAbstractAutocorrection* self, intptr_t slot) {
    auto* vtextautocorrectioncoreimportabstractautocorrection = dynamic_cast<VirtualTextAutoCorrectionCoreImportAbstractAutocorrection*>(self);
    if (vtextautocorrectioncoreimportabstractautocorrection && vtextautocorrectioncoreimportabstractautocorrection->isVirtualTextAutoCorrectionCoreImportAbstractAutocorrection) {
        vtextautocorrectioncoreimportabstractautocorrection->setTextAutoCorrectionCore__ImportAbstractAutocorrection_OperatorAssign_Callback(reinterpret_cast<VirtualTextAutoCorrectionCoreImportAbstractAutocorrection::TextAutoCorrectionCore__ImportAbstractAutocorrection_OperatorAssign_Callback>(slot));
    }
}

void TextAutoCorrectionCore__ImportAbstractAutocorrection_Delete(TextAutoCorrectionCore__ImportAbstractAutocorrection* self) {
    delete self;
}

File: include/restricted-extras-ktextaddons/libimportabstractautocorrection.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");
const TextAutoCorrectionCore__AutoCorrectionUtils__TypographicQuotes = @import("libqt6").TextAutoCorrectionCore__AutoCorrectionUtils__TypographicQuotes;
const importabstractautocorrection_enums = enums;
const std = @import("std");
const Map_constu8_constu8 = std.StringHashMapUnmanaged([]const u8);
const Set_constu8 = std.StringHashMapUnmanaged(void);

/// ### [Upstream resources](https://api.kde.org/legacy/ktextaddons/html/classTextAutoCorrectionCore_1_1ImportAbstractAutocorrection.html)
pub const TextAutoCorrectionCore__ImportAbstractAutocorrection = extern struct {
    /// ### [Upstream resources](https://api.kde.org/legacy/ktextaddons/html/classTextAutoCorrectionCore_1_1ImportAbstractAutocorrection.html)
    ///
    /// The pointer to the underlying Qt C++ object
    ///
    ptr: QtC.TextAutoCorrectionCore__ImportAbstractAutocorrection,

    pub const _is_TextAutoCorrectionCore__ImportAbstractAutocorrection = {};

    /// ### DEPRECATED: Use `new` instead
    ///
    pub const New = new;

    /// Allocate a new TextAutoCorrectionCore::ImportAbstractAutocorrection object in C++ memory
    ///
    pub fn new() TextAutoCorrectionCore__ImportAbstractAutocorrection {
        return .{ .ptr = qtc.TextAutoCorrectionCore__ImportAbstractAutocorrection_new() };
    }

    /// ### DEPRECATED: Use `new2` instead
    ///
    pub const New2 = new2;

    /// Allocate a new TextAutoCorrectionCore::ImportAbstractAutocorrection object in C++ memory
    ///
    /// ## Parameter(s):
    ///
    /// ` param1: TextAutoCorrectionCore__ImportAbstractAutocorrection `
    ///
    pub fn new2(param1: anytype) TextAutoCorrectionCore__ImportAbstractAutocorrection {
        comptime _ = @TypeOf(param1)._is_TextAutoCorrectionCore__ImportAbstractAutocorrection;
        return .{ .ptr = qtc.TextAutoCorrectionCore__ImportAbstractAutocorrection_new2(@ptrCast(param1.ptr)) };
    }

    /// ### DEPRECATED: Use `import` instead
    ///
    pub const Import = import;

    /// ### [Upstream resources](https://api.kde.org/legacy/ktextaddons/html/classTextAutoCorrectionCore_1_1ImportAbstractAutocorrection.html)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: TextAutoCorrectionCore__ImportAbstractAutocorrection `
    ///
    /// ` fileName: []const u8 `
    ///
    /// ` errorMessage: []const u8 `
    ///
    /// ` loadAttribute: importabstractautocorrection_enums.LoadAttribute `
    ///
    pub fn import(self: TextAutoCorrectionCore__ImportAbstractAutocorrection, fileName: []const u8, errorMessage: []const u8, loadAttribute: i32) bool {
        const fileName_str = qtc.libqt_string{
            .len = fileName.len,
            .data = fileName.ptr,
        };
        const errorMessage_str = qtc.libqt_string{
            .len = errorMessage.len,
            .data = errorMessage.ptr,
        };
        return qtc.TextAutoCorrectionCore__ImportAbstractAutocorrection_Import(@ptrCast(self.ptr), fileName_str, errorMessage_str, @bitCast(loadAttribute));
    }

    /// ### DEPRECATED: Use `onImport` instead
    ///
    pub const OnImport = onImport;

    /// ### [Upstream resources](https://api.kde.org/legacy/ktextaddons/html/classTextAutoCorrectionCore_1_1ImportAbstractAutocorrection.html)
    ///
    /// Allows for overriding the related default method
    ///
    /// ## Parameters:
    ///
    /// ` self: TextAutoCorrectionCore__ImportAbstractAutocorrection `
    ///
    /// ` callback: *const fn (self: TextAutoCorrectionCore__ImportAbstractAutocorrection, fileName: [*:0]const u8, errorMessage: [*:0]const u8, loadAttribute: importabstractautocorrection_enums.LoadAttribute) callconv(.c) bool `
    ///
    pub fn onImport(self: TextAutoCorrectionCore__ImportAbstractAutocorrection, callback: *const fn (TextAutoCorrectionCore__ImportAbstractAutocorrection, [*:0]const u8, [*:0]const u8, i32) callconv(.c) bool) void {
        qtc.TextAutoCorrectionCore__ImportAbstractAutocorrection_OnImport(@ptrCast(self.ptr), @bitCast(@intFromPtr(callback)));
    }

    /// ### DEPRECATED: Use `superImport` instead
    ///
    pub const SuperImport = superImport;

    /// ### [Upstream resources](https://api.kde.org/legacy/ktextaddons/html/classTextAutoCorrectionCore_1_1ImportAbstractAutocorrection.html)
    ///
    /// Base class method implementation
    ///
    /// ## Parameter(s):
    ///
    /// ` self: TextAutoCorrectionCore__ImportAbstractAutocorrection `
    ///
    /// ` fileName: []const u8 `
    ///
    /// ` errorMessage: []const u8 `
    ///
    /// ` loadAttribute: importabstractautocorrection_enums.LoadAttribute `
    ///
    pub fn superImport(self: TextAutoCorrectionCore__ImportAbstractAutocorrection, fileName: []const u8, errorMessage: []const u8, loadAttribute: i32) bool {
        const fileName_str = qtc.libqt_string{
            .len = fileName.len,
            .data = fileName.ptr,
        };
        const errorMessage_str = qtc.libqt_string{
            .len = errorMessage.len,
            .data = errorMessage.ptr,
        };
        return qtc.TextAutoCorrectionCore__ImportAbstractAutocorrection_SuperImport(@ptrCast(self.ptr), fileName_str, errorMessage_str, @bitCast(loadAttribute));
    }

    /// ### DEPRECATED: Use `upperCaseExceptions` instead
    ///
    pub const UpperCaseExceptions = upperCaseExceptions;

    /// ### [Upstream resources](https://api.kde.org/legacy/ktextaddons/html/classTextAutoCorrectionCore_1_1ImportAbstractAutocorrection.html)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: TextAutoCorrectionCore__ImportAbstractAutocorrection `
    ///
    /// ` allocator: std.mem.Allocator `
    ///
    pub fn upperCaseExceptions(self: TextAutoCorrectionCore__ImportAbstractAutocorrection, allocator: std.mem.Allocator) Set_constu8 {
        const _set: qtc.libqt_list = qtc.TextAutoCorrectionCore__ImportAbstractAutocorrection_UpperCaseExceptions(@ptrCast(self.ptr));
        var _ret: Set_constu8 = .empty;
        _ret.ensureTotalCapacity(allocator, @intCast(_set.len)) catch @panic("TextAutoCorrectionCore__ImportAbstractAutocorrection.upperCaseExceptions: Total capacity allocation failed");
        const _data_val: [*]qtc.libqt_string = @ptrCast(@alignCast(_set.data));
        for (0.._set.len) |i|
            _ret.putAssumeCapacity(_data_val[i].data[0.._data_val[i].len], {});
        return _ret;
    }

    /// ### DEPRECATED: Use `twoUpperLetterExceptions` instead
    ///
    pub const TwoUpperLetterExceptions = twoUpperLetterExceptions;

    /// ### [Upstream resources](https://api.kde.org/legacy/ktextaddons/html/classTextAutoCorrectionCore_1_1ImportAbstractAutocorrection.html)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: TextAutoCorrectionCore__ImportAbstractAutocorrection `
    ///
    /// ` allocator: std.mem.Allocator `
    ///
    pub fn twoUpperLetterExceptions(self: TextAutoCorrectionCore__ImportAbstractAutocorrection, allocator: std.mem.Allocator) Set_constu8 {
        const _set: qtc.libqt_list = qtc.TextAutoCorrectionCore__ImportAbstractAutocorrection_TwoUpperLetterExceptions(@ptrCast(self.ptr));
        var _ret: Set_constu8 = .empty;
        _ret.ensureTotalCapacity(allocator, @intCast(_set.len)) catch @panic("TextAutoCorrectionCore__ImportAbstractAutocorrection.twoUpperLetterExceptions: Total capacity allocation failed");
        const _data_val: [*]qtc.libqt_string = @ptrCast(@alignCast(_set.data));
        for (0.._set.len) |i|
            _ret.putAssumeCapacity(_data_val[i].data[0.._data_val[i].len], {});
        return _ret;
    }

    /// ### DEPRECATED: Use `autocorrectEntries` instead
    ///
    pub const AutocorrectEntries = autocorrectEntries;

    /// ### [Upstream resources](https://api.kde.org/legacy/ktextaddons/html/classTextAutoCorrectionCore_1_1ImportAbstractAutocorrection.html)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: TextAutoCorrectionCore__ImportAbstractAutocorrection `
    ///
    /// ` allocator: std.mem.Allocator `
    ///
    pub fn autocorrectEntries(self: TextAutoCorrectionCore__ImportAbstractAutocorrection, allocator: std.mem.Allocator) Map_constu8_constu8 {
        const _map: qtc.libqt_map = qtc.TextAutoCorrectionCore__ImportAbstractAutocorrection_AutocorrectEntries(@ptrCast(self.ptr));
        var _ret: Map_constu8_constu8 = .empty;
        _ret.ensureTotalCapacity(allocator, @intCast(_map.len)) catch @panic("TextAutoCorrectionCore__ImportAbstractAutocorrection.autocorrectEntries: Total capacity allocation failed");
        defer {
            const _keys: [*]qtc.libqt_string = @ptrCast(@alignCast(_map.keys));
            const _values: [*]qtc.libqt_string = @ptrCast(@alignCast(_map.values));
            for (0.._map.len) |i| {
                qtc.libqt_free(_keys[i].data);
                qtc.libqt_free(_values[i].data);
            }
            qtc.libqt_free(_map.keys);
            qtc.libqt_free(_map.values);
        }
        const _keys: [*]qtc.libqt_string = @ptrCast(@alignCast(_map.keys));
        const _values: [*]qtc.libqt_string = @ptrCast(@alignCast(_map.values));
        var i: usize = 0;
        while (i < _map.len) : (i += 1) {
            const _key = _keys[i];
            const _entry_slice = allocator.alloc(u8, _key.len) catch @panic("TextAutoCorrectionCore__ImportAbstractAutocorrection.autocorrectEntries: Memory allocation failed");
            @memcpy(_entry_slice, _key.data);
            const _value = _values[i];
            const _value_slice = allocator.alloc(u8, _value.len) catch @panic("TextAutoCorrectionCore__ImportAbstractAutocorrection.autocorrectEntries: Memory allocation failed");
            @memcpy(_value_slice, _value.data);
            _ret.putAssumeCapacity(_entry_slice, _value_slice);
        }
        return _ret;
    }

    /// ### DEPRECATED: Use `superScriptEntries` instead
    ///
    pub const SuperScriptEntries = superScriptEntries;

    /// ### [Upstream resources](https://api.kde.org/legacy/ktextaddons/html/classTextAutoCorrectionCore_1_1ImportAbstractAutocorrection.html)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: TextAutoCorrectionCore__ImportAbstractAutocorrection `
    ///
    /// ` allocator: std.mem.Allocator `
    ///
    pub fn superScriptEntries(self: TextAutoCorrectionCore__ImportAbstractAutocorrection, allocator: std.mem.Allocator) Map_constu8_constu8 {
        const _map: qtc.libqt_map = qtc.TextAutoCorrectionCore__ImportAbstractAutocorrection_SuperScriptEntries(@ptrCast(self.ptr));
        var _ret: Map_constu8_constu8 = .empty;
        _ret.ensureTotalCapacity(allocator, @intCast(_map.len)) catch @panic("TextAutoCorrectionCore__ImportAbstractAutocorrection.superScriptEntries: Total capacity allocation failed");
        defer {
            const _keys: [*]qtc.libqt_string = @ptrCast(@alignCast(_map.keys));
            const _values: [*]qtc.libqt_string = @ptrCast(@alignCast(_map.values));
            for (0.._map.len) |i| {
                qtc.libqt_free(_keys[i].data);
                qtc.libqt_free(_values[i].data);
            }
            qtc.libqt_free(_map.keys);
            qtc.libqt_free(_map.values);
        }
        const _keys: [*]qtc.libqt_string = @ptrCast(@alignCast(_map.keys));
        const _values: [*]qtc.libqt_string = @ptrCast(@alignCast(_map.values));
        var i: usize = 0;
        while (i < _map.len) : (i += 1) {
            const _key = _keys[i];
            const _entry_slice = allocator.alloc(u8, _key.len) catch @panic("TextAutoCorrectionCore__ImportAbstractAutocorrection.superScriptEntries: Memory allocation failed");
            @memcpy(_entry_slice, _key.data);
            const _value = _values[i];
            const _value_slice = allocator.alloc(u8, _value.len) catch @panic("TextAutoCorrectionCore__ImportAbstractAutocorrection.superScriptEntries: Memory allocation failed");
            @memcpy(_value_slice, _value.data);
            _ret.putAssumeCapacity(_entry_slice, _value_slice);
        }
        return _ret;
    }

    /// ### DEPRECATED: Use `typographicSingleQuotes` instead
    ///
    pub const TypographicSingleQuotes = typographicSingleQuotes;

    /// ### [Upstream resources](https://api.kde.org/legacy/ktextaddons/html/classTextAutoCorrectionCore_1_1ImportAbstractAutocorrection.html)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: TextAutoCorrectionCore__ImportAbstractAutocorrection `
    ///
    pub fn typographicSingleQuotes(self: TextAutoCorrectionCore__ImportAbstractAutocorrection) TextAutoCorrectionCore__AutoCorrectionUtils__TypographicQuotes {
        return .{ .ptr = qtc.TextAutoCorrectionCore__ImportAbstractAutocorrection_TypographicSingleQuotes(@ptrCast(self.ptr)) };
    }

    /// ### DEPRECATED: Use `typographicDoubleQuotes` instead
    ///
    pub const TypographicDoubleQuotes = typographicDoubleQuotes;

    /// ### [Upstream resources](https://api.kde.org/legacy/ktextaddons/html/classTextAutoCorrectionCore_1_1ImportAbstractAutocorrection.html)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: TextAutoCorrectionCore__ImportAbstractAutocorrection `
    ///
    pub fn typographicDoubleQuotes(self: TextAutoCorrectionCore__ImportAbstractAutocorrection) TextAutoCorrectionCore__AutoCorrectionUtils__TypographicQuotes {
        return .{ .ptr = qtc.TextAutoCorrectionCore__ImportAbstractAutocorrection_TypographicDoubleQuotes(@ptrCast(self.ptr)) };
    }

    /// ### DEPRECATED: Use `maxFindStringLenght` instead
    ///
    pub const MaxFindStringLenght = maxFindStringLenght;

    /// ### [Upstream resources](https://api.kde.org/legacy/ktextaddons/html/classTextAutoCorrectionCore_1_1ImportAbstractAutocorrection.html)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: TextAutoCorrectionCore__ImportAbstractAutocorrection `
    ///
    pub fn maxFindStringLenght(self: TextAutoCorrectionCore__ImportAbstractAutocorrection) i32 {
        return qtc.TextAutoCorrectionCore__ImportAbstractAutocorrection_MaxFindStringLenght(@ptrCast(self.ptr));
    }

    /// ### DEPRECATED: Use `minFindStringLenght` instead
    ///
    pub const MinFindStringLenght = minFindStringLenght;

    /// ### [Upstream resources](https://api.kde.org/legacy/ktextaddons/html/classTextAutoCorrectionCore_1_1ImportAbstractAutocorrection.html)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: TextAutoCorrectionCore__ImportAbstractAutocorrection `
    ///
    pub fn minFindStringLenght(self: TextAutoCorrectionCore__ImportAbstractAutocorrection) i32 {
        return qtc.TextAutoCorrectionCore__ImportAbstractAutocorrection_MinFindStringLenght(@ptrCast(self.ptr));
    }

    /// ### DEPRECATED: Use `operatorAssign` instead
    ///
    pub const OperatorAssign = operatorAssign;

    /// ### [Upstream resources](https://api.kde.org/legacy/ktextaddons/html/classTextAutoCorrectionCore_1_1ImportAbstractAutocorrection.html)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: TextAutoCorrectionCore__ImportAbstractAutocorrection `
    ///
    /// ` param1: TextAutoCorrectionCore__ImportAbstractAutocorrection `
    ///
    pub fn operatorAssign(self: TextAutoCorrectionCore__ImportAbstractAutocorrection, param1: anytype) void {
        comptime _ = @TypeOf(param1)._is_TextAutoCorrectionCore__ImportAbstractAutocorrection;
        qtc.TextAutoCorrectionCore__ImportAbstractAutocorrection_OperatorAssign(@ptrCast(self.ptr), @ptrCast(param1.ptr));
    }

    /// ### DEPRECATED: Use `onOperatorAssign` instead
    ///
    pub const OnOperatorAssign = onOperatorAssign;

    /// ### [Upstream resources](https://api.kde.org/legacy/ktextaddons/html/classTextAutoCorrectionCore_1_1ImportAbstractAutocorrection.html)
    ///
    /// Allows for overriding the related default method
    ///
    /// ## Parameters:
    ///
    /// ` self: TextAutoCorrectionCore__ImportAbstractAutocorrection `
    ///
    /// ` callback: *const fn (self: TextAutoCorrectionCore__ImportAbstractAutocorrection, param1: TextAutoCorrectionCore__ImportAbstractAutocorrection) callconv(.c) void `
    ///
    pub fn onOperatorAssign(self: TextAutoCorrectionCore__ImportAbstractAutocorrection, callback: *const fn (TextAutoCorrectionCore__ImportAbstractAutocorrection, TextAutoCorrectionCore__ImportAbstractAutocorrection) callconv(.c) void) void {
        qtc.TextAutoCorrectionCore__ImportAbstractAutocorrection_OnOperatorAssign(@ptrCast(self.ptr), @bitCast(@intFromPtr(callback)));
    }

    /// ### DEPRECATED: Use `superOperatorAssign` instead
    ///
    pub const SuperOperatorAssign = superOperatorAssign;

    /// ### [Upstream resources](https://api.kde.org/legacy/ktextaddons/html/classTextAutoCorrectionCore_1_1ImportAbstractAutocorrection.html)
    ///
    /// Base class method implementation
    ///
    /// ## Parameter(s):
    ///
    /// ` self: TextAutoCorrectionCore__ImportAbstractAutocorrection `
    ///
    /// ` param1: TextAutoCorrectionCore__ImportAbstractAutocorrection `
    ///
    pub fn superOperatorAssign(self: TextAutoCorrectionCore__ImportAbstractAutocorrection, param1: anytype) void {
        comptime _ = @TypeOf(param1)._is_TextAutoCorrectionCore__ImportAbstractAutocorrection;
        qtc.TextAutoCorrectionCore__ImportAbstractAutocorrection_SuperOperatorAssign(@ptrCast(self.ptr), @ptrCast(param1.ptr));
    }

    /// ### DEPRECATED: Use `delete` instead
    ///
    pub const Delete = delete;

    /// ### [Upstream resources](https://api.kde.org/legacy/ktextaddons/html/classTextAutoCorrectionCore_1_1ImportAbstractAutocorrection.html)
    ///
    /// Delete this object from C++ memory
    ///
    /// ## Parameter:
    ///
    /// ` self: TextAutoCorrectionCore__ImportAbstractAutocorrection `
    ///
    pub fn delete(self: TextAutoCorrectionCore__ImportAbstractAutocorrection) void {
        qtc.TextAutoCorrectionCore__ImportAbstractAutocorrection_Delete(@ptrCast(self.ptr));
    }
};

pub const enums = struct {
    pub const LoadAttribute = enum(i32) {
        pub const All: i32 = 0;
        pub const SuperScript: i32 = 1;
    };
};
